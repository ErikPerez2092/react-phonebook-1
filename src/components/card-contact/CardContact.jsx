import React from 'react';
import PropTypes from 'prop-types';
import FlatButton from 'material-ui/FlatButton';
import './cardContact.css';

const CardContact = ({ id, img, name, nickname, group, handleClick }) => (
  <FlatButton className="CardContact" fullWidth={true} onClick={() => handleClick(id)}>
    <div className="CardContact-img" style={{ backgroundImage: `url(${img})` }} />
    <h2 className="CardContact-name">{name}</h2>
    <h3 className="CardContact-nickname">{nickname}</h3>
    <span className="CardContact-group">{group}</span>
  </FlatButton>
);

CardContact.propTypes = {
  id: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  nickname: PropTypes.string.isRequired,
  group: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default CardContact;
