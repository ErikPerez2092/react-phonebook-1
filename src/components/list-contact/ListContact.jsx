import React from 'react';
import PropTypes from 'prop-types';

import CardContact from '../card-contact';

const ListContact = ({ contacts }) => contacts.map(contact => <CardContact key={contact.id} {...contact} />);

ListContact.propTypes = {
  contacts: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      img: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      nickname: PropTypes.string.isRequired,
      group: PropTypes.string.isRequired,
    }),
  ),
};

export default ListContact;
