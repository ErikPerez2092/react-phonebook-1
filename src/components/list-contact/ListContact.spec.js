import React from 'react';
import { mount } from 'enzyme';
import ListContact from './ListContact';
import CardContact from '../card-contact/CardContact';

describe('ListContactComponent', () => {
  let listContactComponent;
  const contacts = [
    {
      id: '1',
      img: 'https://randomuser.me/api/portraits/men/83.jpg',
      name: 'Edgar Zavala',
      nickname: 'ezavala',
      group: 'friends',
    },
    {
      id: '2',
      img: 'https://randomuser.me/api/portraits/men/84.jpg',
      name: 'Juan Zavala',
      nickname: 'jazavala',
      group: 'friends',
    },
  ];

  beforeEach(() => {
    listContactComponent = mount(<ListContact contacts={contacts} />);
  });

  it('sould render <ListContact />', () => {
    expect(listContactComponent.getElements()).toMatchSnapshot();
  });

  it('should contain two <CardContact />', () => {
    expect(listContactComponent.find(CardContact)).toHaveLength(2);
  });
});
