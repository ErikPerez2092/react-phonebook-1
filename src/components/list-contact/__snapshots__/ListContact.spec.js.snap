// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListContactComponent sould render <ListContact /> 1`] = `
Array [
  <ListContact
    contacts={
        Array [
            Object {
              "group": "friends",
              "id": "1",
              "img": "https://randomuser.me/api/portraits/men/83.jpg",
              "name": "Edgar Zavala",
              "nickname": "ezavala",
            },
            Object {
              "group": "friends",
              "id": "2",
              "img": "https://randomuser.me/api/portraits/men/84.jpg",
              "name": "Juan Zavala",
              "nickname": "jazavala",
            },
          ]
    }
/>,
]
`;
